WcaOnRails/spec/controllers/registrations_controller_spec.rb

```diff
@@ -133,6 +133,29 @@
       expect(accepted_registration.reload.accepted?).to be true
     end
 
+    it "doesn't allow undeleting a banned user while allowing actions on the others" do
+      registration.update!(accepted_at: Time.now)
+      registration2 = FactoryBot.create(:registration, :pending, competition: competition)
+      deleted_registration = FactoryBot.create(:registration, :deleted, competition: competition)
+      banned_deleted_registration = FactoryBot.create(:registration, :deleted, competition: competition)
+      banned_user = FactoryBot.create(:user, :banned)
+      banned_deleted_registration.update!(user: banned_user)
```

Comment:
You can pass the user when creating the registration, like so:
```irb
FactoryBot.create(:registration, :deleted, competition: competition, user: FactoryBot.create(:user, :banned))
```

Reply:
That's a nice shortcut! However creating a registration for a banned user fails validation with `User You are banned` etc.